File: python/21a_996A_HitTheLottery.py
def main():
    n = int(input())
    temp = n
    cnt = 0
    while temp > 0:
        if temp - 100 >= 0:
            temp -= 100
            cnt += 1
        elif temp - 20 >= 0:
            temp -= 20
            cnt += 1
        elif temp - 10 >= 0:
            temp -= 10
            cnt += 1
        elif temp - 5 >= 0:
            temp -= 5
            cnt += 1
        elif temp - 1 >= 0:
            temp -= 1
            cnt += 1
    print(cnt)

if __name__ == "__main__":
    main()

File: cpp/21a_996A_HitTheLottery.cpp
#include<bits/stdc++.h>

int main(){
    int n, cnt = 0, temp;
    std::cin>>n;
    temp = n;
    while(temp > 0){
        if(temp - 100 >= 0){
            temp -= 100;
            cnt++;
        }
        else if(temp - 20 >= 0){
            temp -= 20;
            cnt++;
        }
        else if(temp - 10 >= 0){
            temp -= 10;
            cnt++;
        }
        else if(temp - 5 >= 0){
            temp -= 5;
            cnt++;
        }
        else if(temp - 1 >= 0){
            temp -= 1;
            cnt ++;
        }
    }
    std::cout<<cnt;
    return 0;
}